libs/core/execution_base/include/hpx/execution_base/completion_signatures.hpp

```diff
@@ -703,6 +715,615 @@ namespace hpx::execution::experimental {
+                    std::rethrow_exception(std::get<2>(result));
+                }
+                std::terminate();
+            }
+
+        protected:
+            coroutine_expected_result_t<Value> result;
+        };
+
+        template <typename PromiseId, typename SenderId>
+        struct sender_awaitable;
+
+        template <typename Promise, typename Sender>
+        using sender_awaitable_t = sender_awaitable<hpx::meta::hidden<Promise>,
+            hpx::meta::hidden<Sender>>;
+
+        template <typename Sender, typename Promise>
+        using receiver =
+            typename sender_awaitable_base<hpx::meta::hidden<Promise>,
+                single_sender_value_t<Sender, env_of_t<Promise>>>::receiver;
+
+        template <typename Sender, typename Env = no_env>
+        inline constexpr bool is_single_typed_sender_v =
+            is_sender_v<Sender, Env>&& hpx::meta::value<
+                hpx::meta::is_valid<single_sender_value_t, Sender, Env>>;
+
+        template <typename Sender, typename Promise>
+        inline constexpr bool is_awaitable_sender_v =
+            is_single_typed_sender_v<Sender, env_of_t<Promise>>&&
+                is_sender_to_v<Sender, receiver<Sender, Promise>>&&
+                    has_unhandled_stopped<Promise>;
+
+    }    // namespace detail
+
+    inline constexpr struct as_awaitable_t
+      : hpx::functional::detail::tag_fallback<as_awaitable_t>
+    {
+        // static to call this function without creating an object
+        template <typename T, typename Promise>
+        static constexpr bool is_noexcept() noexcept
+        {
+            if constexpr (is_awaitable_v<T>)
+            {
+                return true;
+            }
+            else if constexpr (detail::is_awaitable_sender_v<T, Promise>)
+            {
+                using Sender = detail::sender_awaitable_t<Promise, T>;
+                return std::is_nothrow_constructible_v<Sender, T,
+                    hpx::coro::coroutine_handle<Promise>>;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        template <typename T, typename Promise>
+        friend HPX_FORCEINLINE decltype(auto) tag_fallback_invoke(
+            as_awaitable_t, T&& t,
+            Promise& promise) noexcept(is_noexcept<T, Promise>())
+        {
+            if constexpr (is_awaitable_v<T>)
+            {
+                return HPX_FORWARD(T, t);
+            }
+            else if constexpr (detail::is_awaitable_sender_v<T, Promise>)
+            {
+                auto hcoro =
+                    hpx::coro::coroutine_handle<Promise>::from_promise(promise);
+                return detail::sender_awaitable_t<Promise, T>{
+                    HPX_FORWARD(T, t), hcoro};
+            }
+            else
+            {
+                return HPX_FORWARD(T, t);
+            }
+        }
+    } as_awaitable{};
+
+    namespace detail {
+        struct with_awaitable_senders_base
```

Comment:
Codacy is complaining about this type having private data members but no constructors. I know that this is a false positive and I have tried to convince it to ignore the issue. Could you please add a 
```
    with_awaitable_senders_base() = default;
```
to this class to silence the warning?